CMakeLists.txt

```diff
@@ -112,7 +129,9 @@ check_include_file(sys/resource.h   HAVE_SYS_RESOURCE_H) # for getrusage
 check_include_file("dlfcn.h"        HAVE_DLFCN_H)
 check_include_file("endian.h"       HAVE_ENDIAN_H)
 check_include_file("limits.h"       HAVE_LIMITS_H)
-check_include_file("locale.h"       HAVE_LOCALE_H)
+if (NOT AMIGA)
+    check_include_file("locale.h"       HAVE_LOCALE_H)
```

Comment:
overall the changes look ok, but I'm wondering why do you add explicit guards around some of these checks?  Does amiga have these headers and symbols, but they don't work as expected?

Reply:
<img width="675" alt="image" src="https://github.com/json-c/json-c/assets/6903516/c3cce8f5-7ebf-4453-90db-b41b9a4676f5">

Amiga has these headers but a lot of the functions json-c uses from them are not available unless you are compiling a POSIX compliant app.

Reply:
Wow, that's pretty broken of those libraries, but ISTM that the right way to work around isn't to add "AMIGA" special checks, but instead check for the symbols in addition to the headers.  In fact, we already do so with e.g. the `HAVE_USELOCALE` define: https://github.com/json-c/json-c/blob/master/CMakeLists.txt#L175
Can you figure out why HAVE_USELOCALE ends up defined if the symbol isn't actually available?

Reply:
Thanks for your feedback. I'll look into this :)

In the meantime I have already released a compiled library at http://aminet.net/package/dev/lib/json-c

But for sure I will work to make the cmake build as clean as possible to be accepted into this repo :)

Reply:
Ok after much frustration I discovered that the current version of the C runtime libraries haven't actually implemented all the functions despite them being defined in the header files. At least this is the case for the Motorola 68k. The PowerPC libraries are more complete and don't require such checks.

Reply:
hmm... I wonder if we're running into issues here because you're cross-compiling, and cmake isn't using the right toolchain for the check_symbol_exist steps.  I notice that your readme instructions don't mention passing a `--toolchain` option as suggested at https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html#cross-compiling
Can you perhaps try that and see if it improves the situation?

Reply:
The correct toolchain automatically gets used because its set in an environment variable baked in the Amiga gcc Docker images

Reply:
Looking at the docs for check_symbol_exists, it says `If the header files declare the symbol as a function or variable then the symbol must also be available for linking`

Is `uselocale` defined as a macro in your toolchain header files?  If not, then I would expect check_symbol_exists to be able to figure out that the symbol doesn't actually exist in the library.
If you add something like this, does cmake think the symbol exists in the library?
```
check_function_exists(uselocale HAVE_USELOCALE_IN_LIBRARY)
if (HAVE_USELOCALE_IN_LIBRARY)
message(STATUS "uselocale() IS present in library")
else()
message(STATUS "uselocale() is NOT present in library")
endif()
```

I think we need to figure out why the check isn't detecting the lack of the symbols in the actual library.  Then, if it turns out we do need some kind of exception, it should be down after `HAVE_USELOCAL`/`HAVE_SETLOCALE` are set, similar to the special case for uClibc (probably using the check_function_exists macro)

Reply:
OK so I removed the Amiga specific guard on locale.h because I figured out why it was broken when you want to build the clib2 version of json-c. Turns out there was a bug in clib2 where you will get an error about conflicting types if your project has an `#include` for both `locale.h` and `math.h` and you're building from Bebbo's Amiga dev toolchain. I have created a PR https://github.com/adtools/clib2/pull/18 to fix that issue. Now there are no locale issues when building json-c. Basically clib2 just didn't properly cater for other c runtime libraries in the environment other than itself. Also it seems newlib has been updated to add definitions for `use locale()`, `freelocale()` and `duplocale()` so those linking errors in the screenshot I posted a few months ago are no longer occurring.

Reply:
Awesome work figuring this out, thank you!
I'll review and merge this in the next week or so.